.github/workflows/ci.yml

```diff
@@ -10,14 +10,38 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-    
-      - name: Set up Python 
+
+      - name: Set up Python
         uses: actions/setup-python@v4
         with:
           python-version: "3.x"
-      
-      - name: Install pypa/build
-        run: python3 -m pip install build poetry --user
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+      - name: Check
+        run: |
+            poetry run ruff check .
+            poetry run black --check .
+
+      - name: Install
+        run: poetry install
 
-      - name: Test the build 
-        run: python3 -m build
+      - name: Build
+        run: poetry build
+  documentation:
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags')
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+      - name: Build site
+        run: |
+          poetry install
+          poetry run mkdocs build -f documentation/mkdocs-material.yml
+      - name: Build and Deploy
+        uses: JamesIves/github-pages-deploy-action@4.7.2
+        with:
+          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+          BRANCH: gh-pages
+          FOLDER: documentation/site
```

Comment:
Same goes for this one. Lets use the official GitHub Action for deploying pages:
https://github.com/actions/deploy-pages